mbtools/models.py

```diff
@@ -99,18 +142,19 @@ def html_elements(self):
                 f"//question[@id={self.id}]/questiontext")
         for question_html in question_texts:
             elements.append(
-                MoodleHtmlElement(question_html))
+                MoodleHtmlElement(question_html, self.location))
         answer_texts = self.etree.xpath(
                 f"//question[@id={self.id}]/answers/answer/answertext")
         for answer_html in answer_texts:
             elements.append(
-                MoodleHtmlElement(answer_html))
+                MoodleHtmlElement(answer_html, self.location))
         return elements
 
 
 class MoodleHtmlElement:
-    def __init__(self, parent):
+    def __init__(self, parent, location):
```

Comment:
No, wait, I was right the first time I think :sweat_smile: . `MoodleHtmlElement` can have different types of parents so it can't know for sure what to parse up the tree, but it can incorporate a combination of its parent's `location_string()` along with information it can parse about itself. 

Maybe we can pair for a bit tomorrow to discuss / think about this a bit more.

Reply:
That would be great, It'd be helpful to ask a bunch of questions👍